onnxruntime/core/mlas/inc/mlas_qnbit.h

```diff
@@ -77,3 +77,119 @@ MlasIsSQNBitGemmAvailable(
+
+/**
+ * @brief Define compute types of block quantization
+ */
+typedef enum {
+    CompUndef = 0, /*!< undef */
+    CompFp32 = 1,  /*!< input fp32, accumulator fp32 */
+    CompFp16 = 2,  /*!< input fp16, accumulator fp16 */
+    CompBf16 = 3,  /*!< input bf16, accumulator fp32 */
+    CompInt8 = 4   /*!< input int8, accumulator int32 */
+} MLAS_COMPUTE_TYPE;
+
+/**
+ * @brief Data parameters for NBits GEMM routine
+ *        C = A * B
+ *        A, C must be a float32 matrix
+ *        B must be a packed nbits blob
+ *        All except C are [in] parameters
+ */
+struct MLAS_NBITS_GEMM_DATA_PACKED_PARAMS {
+    const float* A = nullptr; /**< address of A (float32 matrix)*/
+    const void* B = nullptr;  /**< address of B (packed nbits blob)*/
+    float* C = nullptr;       /**< address of result matrix */
+    size_t lda = 0;           /**< leading dimension of A */
+    size_t ldc = 0;           /**< leading dimension of C*/
+};
+
+/**
+ * @brief Compute the byte size of the parameter combination
+ *
+ * @param N      the number of columns of matrix B.
+ * @param K      the number of rows of matrix B.
+ * @param block_size    size of the block to quantize, elements from the same block share the same
+ * scale and zero point
+ * @param nbits  number of bits used for weight quantization (default 4)
+ * @param is_asym  flag for asymmetric quantization
+ * @param comp_type  specify input data type and accumulator data type
+ * @return size of the packing buffer, 0 if the operation is not yet supported.
+ */
+size_t MLASCALL
+MlasNBitsGemmPackBSize(size_t N, size_t K, size_t block_size, int nbits, bool is_asym, MLAS_COMPUTE_TYPE comp_type);
+
+/**
+ * @brief Prepack tensor data from MatMulNBits operator
+ *
+ * @param PackedBuf     pakced data buffer
+ * @param QData         quantized data buffer
+ * @param Scale         scale pointer
+ * @param Zp            zero point pointer
+ * @param N             the number of columns of matrix B.
+ * @param K             the number of rows of matrix B.
+ * @param ldb           leading dimension of B
+ * @param block_size    size of the block to quantize, elements from the same block share the same
+ * scale and zero point
+ * @param nbits         number of bits used for weight quantization (default 4)
+ * @param is_asym       flag for asymmetric quantization
+ * @param comp_type     specify input data type and accumulator data type
+ * @param last_call     flag to activate the epilogue process of packB
+ * @param thread_pool
+ */
+void MLASCALL
+MlasNBitsGemmPackB(
+    void* PackedBuf,
+    const uint8_t* QData,
+    const float* Scale,
+    const uint8_t* Zp,
+    size_t N,
+    size_t K,
+    size_t ldb,
+    size_t block_size,
+    int nbits,
+    bool is_asym,
+    bool last_call,
+    MLAS_COMPUTE_TYPE comp_type,
+    MLAS_THREADPOOL* thread_pool
+);
+
+/**
+ * @brief Unpack and dequantize to fp32
+ *
+ * @param FpData     unpakced float32 data
+ * @param PackedBuf  int4 quantized and packed data
```

Comment:
"int4" - is this function intended to also support other values of `nbits`?

Reply:
Yes, it's intended to support other values. 'int4' has been removed from param doc.